tmpl/conversions_windows.tmpl

```diff
@@ -0,0 +1,54 @@
+//glow:keepspace
+// +build windows
+
+// Code generated by glow (https://github.com/go-gl/glow). DO NOT EDIT.
+
+package {{.Name}}
+
+import (
+	"runtime"
+	"strings"
+	"unsafe"
+)
+
+// GoStr takes a null-terminated string returned by OpenGL and constructs a
+// corresponding Go string.
+func GoStr(cstr *uint8) string {
+	str := ""
+	for {
+		if *cstr == 0 {
+			break
+		}
+		str += string(*cstr)
+		cstr = (*uint8)(unsafe.Pointer(uintptr(unsafe.Pointer(cstr)) + 1))
+	}
+	return str
+}
+
+// Strs takes a list of Go strings (with or without null-termination) and
+// returns their C counterpart.
+//
+// The returned free function must be called once you are done using the strings
+// in order to free the memory.
+//
+// If no strings are provided as a parameter this function will panic.
+func Strs(strs ...string) (cstrs **uint8, free func()) {
```

Comment:
@errcw My concern was that `**uint8` would be a Go pointer to a Go pointer, which is not allowed to pass to C world. (This is fine to pass this to `Syscall` on Windows as long as we care the lifetime of the object). That's why I separated the implementation for Windows and non-Windows. Do you have any suggestions?

Reply:
ping

Reply:
I'm so sorry for the slow responses here. The amount of free time I have is vanishingly small these days. The `cstrs **unit8` return value is actually a set of pointers into malloc'ed C memory, which I believe is safe?

Reply:
Thank you for responding! Our plan is to make this function in pure Go (no cgo in any environment), right? If so, we would not be able to use C allocator.

Reply:
Right, right, of course. So I'm clear, on Windows, we're using `Syscall` which is inherently safe to accept values that are Go pointers to Go pointers? Whereas on other platforms those same values are unsafe because they cross the Cgo boundary? If so, sounds good.

Reply:
> on Windows, we're using Syscall which is inherently safe to accept values that are Go pointers to Go pointers?

Yes, there is no warning, and we can assure the lifetime of the objects.

> Whereas on other platforms those same values are unsafe because they cross the Cgo boundary?

This is unsafe, and unfortunately Go runtime suspends the execution with warning messages. That's the problem.

Then, I think it would be inevitable to split the implementation for Windows and non-Windows. My suggestion is to split the allocation part to minimize the split code. What do you think?

Reply:
>  If so, sounds good.

So, you are fine with splitting the code? :-)